lib/safe_macros.c

```diff
@@ -15,6 +15,7 @@
 #include <stdlib.h>
 #include <unistd.h>
 #include <malloc.h>
+#include <stdbool.h>
```

Comment:
The stdbool is only useful when we are exporting interfaces in a header file. It was created after many project redefined bool in headers which caused redefinitions and conflicts. There is no point in using bool for something that is used only inside a single function...